### ISSUE TITLE
Interpolation onto irregular data locations

### ISSUE THREAD

**Issue by JianghuiDu** [gher-ulg/DIVAnd.jl#50]
The ODV docs mention that it can do 3D estimates at arbitrary locations using DIVA gridding results, is there a way to do this using DIVAnd? The problem is quite simple: given a 3D gridded ocean field, estimate the value at certain locations (x,y,z).

**Comment by jmbeckers** [comment]
If the locations are those of the data used for the analysis you can use DIVAnd_residualobs to get the difference between the gridded field and the data (so you can get the gridded field at the location too).

If you know in advance (before you run DIVAnd) where you want to look at and the point is not a data point, you can still use DIVAnd_residualobs by adding some pseudo observations at these locations (with an infinite error variance).

If you already did the gridding and want to look at a position not falling on a grid point, you can simply use standard interpolation functions since now your gridded field is "smooth" enough to allow for simple interpolations. Eg. use package Interpolations (already used when working with DIVAnd).

**Comment by JianghuiDu** [comment]
I guess my problem fall somewhere between the 2nd and 3rd. I downloaded a gridded product from some other analysis. I want to compare to my measurements at certain locations which were not used to create the gridded product. Should I use the gridded data as input to perform a DIVAnd analysis, at the same time adding my data but with inf errors, and then do DIVAnd_residualobs? I have tried Interpolations  but it doesn't deal with masks as far as I know, and I'd like to use distance based weights to the interpolation, rather than using linear/cubic methods.
